_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+AGREE
+phy >> TxFrameStartNtf:INFORM[txTime:100610682 txDuration:1000416]
+phy >> TxFrameNtf:INFORM[txTime:100593099]
+```
+You should hear a tone sent out by the Unet audio running in the laptop.
+
+Passband PSD plot from the diagnostics scope shows this carrier only signal at nearly 12kHz.
+
+(settings used: plvl=-25, plot trigger set to -60, you can experiment with power and trigger to get a snapshot depending on your environment)
+
+<p align="center"><img src="../assets/img/freqBandControl/tone.png" style="zoom:50%;"/></p>
+
+The following will send out a 1s BPSK signal with constellation (-1, 1) at the highest rate possible, i.e., every alternating complex sample is (+1,0) followed by (-1,0). 
+
+```
+> s = [1,0, -1, 0]*6000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 6kHz and 18kHz implying full use of the band 6 to 18kHz.
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-high.png" style="zoom:50%;"/></p>
+
+
+The following will send out a 1s BPSK signal at a half the highest rate possible. 
+
+```
+> s = [1,0, 1, 0, -1, 0, -1, 0]*3000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 9kHz and 15kHz implying use of 6kHz bandwidth
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-low.png" style="zoom:50%;"/></p>
+
+Note that in the strict sense, unless a band limiting RRC (Raised Root Cosine) filter or equivalent is used, the effective bandwidth does not reduce correctly for BPSK with random binary data. The proper implementation of BPSK or general PSK is not aimed in this blog.
+
+The above is intended to give the reader a flavour of how frequency band used changes with signals. The reader no doubt will be interested in testing different types of modulation schemes, and using the PSD scope, the frequency band used can be visualized. Surely it can also be estimated from communication theory.
+
+## OFDM frequency band control
+
+OFDM is one of the main modulation schemes provided with Subnero modems. Controlling the bandwidth and centre of the frequency band used is very simple for this scheme.
+
+OFDM uses a number of carriers to modulate the baseband data and the number is specified via `phy[2].nc`. These carriers span the total available bandwidth (24kHz for Medium Frequency Subnero modem variants), i.e., from 12kHz to 36kHz. 
+
+The parameter `phy[2].bw`. controls a reduction in the usage of this total bandwidth. I.e., if we specifiy `phy[2].bw=0.5`, it reduces the bandwidth used by half, i.e, 24kHz/2 = 12kHz. 
+
+The other parameter involved in bandwidth control is `phy[2].dc0`. This controls the starting carrier number. 
+
+`dc0 >= -nc/2 || dc0 <= nc/2`
```

Comment:
Write proper math, rather than math that looks like code.

Reply:
You should be able to use LaTeX. This seems broken at present, but I'll look at fixing it, so go ahead and use it if needed.

Reply:
LaTeX was fixed in https://github.com/org-arl/unet-blog/commit/157d47b3cb46fcc42ac278f8ee93e134791216bb

Reply:
The fix in the master was rebased to this branch. Still, it is not rendering properly in github: https://github.com/org-arl/unet-blog/blob/frequency/_posts/2021-04-21-frequency-band-control-in-unetstack.md


Is this an issue with github and can we simply merge the branches?

Reply:
It should render when you rebase to `master`